hippunfold/workflow/rules/atlas_gen.smk

```diff
@@ -0,0 +1,332 @@
+            bids(
+                root=work,
+                datatype="anat",
+                suffix="{metric}.nii.gz",
+                space="unfold2d",
+                hemi="{hemi}",
+                label="{label}",
+                **inputs.subj_wildcards,
+            ),
+            metric=["gyrification", "curvature", "thickness"],
+            allow_missing=True,
+        ),
+        avgtemplate_dir="template/avgtemplate_{label}",
+    params:
+        glob_input_warp=lambda wildcards, input: "{avgtemplate_dir}/input*-{filename}-1Warp.nii.gz".format(
+            avgtemplate_dir=input.avgtemplate_dir,
+            filename=Path(input.metric_nii[0]).name.removesuffix(".nii.gz"),
+        ),
+        glob_input_invwarp=lambda wildcards, input: "{avgtemplate_dir}/input*-{filename}-1InverseWarp.nii.gz".format(
+            avgtemplate_dir=input.avgtemplate_dir,
+            filename=Path(input.metric_nii[0]).name.removesuffix(".nii.gz"),
+        ),
+    output:
+        warp=bids(
+            root=work,
+            datatype="warps",
+            suffix="warp.nii.gz",
+            from_="unfold",
+            to="avgunfold",
+            hemi="{hemi}",
+            label="{label}",
+            **inputs.subj_wildcards,
+        ),
+        invwarp=bids(
+            root=work,
+            datatype="warps",
+            suffix="invwarp.nii.gz",
+            from_="unfold",
+            to="avgunfold",
+            hemi="{hemi}",
+            label="{label}",
+            **inputs.subj_wildcards,
+        ),
+    group:
+        "subj"
+    shell:
+        "cp {params.glob_input_warp} {output.warp} && "
+        "cp {params.glob_input_invwarp} {output.invwarp}"
+
+
+rule copy_avgtemplate_metric:
+    """ copy avgtemplate metric, adjusting header to match the original data
+     (since this seems to get garbled in z by ants)"""
+    input:
+        avgtemplate_dir="template/avgtemplate_{label}",
+        ref_metric=lambda wildcards: inputs[config["modality"]].expand(
+            bids(
+                root=work,
+                datatype="anat",
+                suffix="{metric}.nii.gz",
+                space="unfold2d",
+                hemi="{hemi}",
+                label="{label}",
+                **inputs.subj_wildcards,
+            ),
+            label=wildcards.label,
+            metric=wildcards.metric,
+            **expand_hemi(),
+        )[0],
+    params:
+        in_metric=lambda wildcards, input: "{avgtemplate_dir}/template{i}.nii.gz".format(
+            avgtemplate_dir=input.avgtemplate_dir,
+            i=reg_metrics.index(wildcards.metric),
+        ),
+    output:
+        metric="template/avgtemplate_label-{label}_{metric}.nii.gz",
+    script:
+        "../scripts/set_metric_nii_header.py"
+
+
+def get_metric_template(wildcards, input):
+
+    num_modalities = len(reg_metrics)
+
+    warp_prefix = "template/warp_label-{label}_"
+    metrics = sorted(glob(f"{warp_prefix}_template?.nii.gz"))
```

Comment:
the problem here is would be globbing before the files exist.. we can instead just point to the output files created in the `copy_avgtemplate_metric` rule. I'll update this..